### PR TITLE
Create 2017-grid-game.js

### PR DESCRIPTION
**Pull request by aadil42**
Solved grid-game in JS.

File(s) Added: 2017-grid-game.js
Language(s) Used: JavaScript
Submission URL: https://leetcode.com/problems/grid-game/submissions/985274543/

### CODE REVIEW

File: javascript/2017-grid-game.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/**
+ * https://leetcode.com/problems/grid-game/
+ * Linear
+ * Time O(n) | Space O(n)
+ * @param {number[][]} grid
+ * @return {number}
+ */
+var gridGame = function(grid) {
+
+    const preSum1 = [0];
```
Comment (aakhtar3) [inline javascript/2017-grid-game.js:10]: Suggestion: break up into modular functions

```js
main = () => {
    preCompute()

    return search()
}
```

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/2017-grid-game.js (hunk 1 above)
Comment (aadil42) [reply javascript/2017-grid-game.js]: I have split them into modules please have a look.

Comment (aadil42) [reply javascript/2017-grid-game.js]: Just a kind reminder that I updated the code as suggested.

File: javascript/2017-grid-game.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        const top = arr1[arr1.length - 1] - arr1[i];
+        const bottom = arr2[i - 1];
+        let secondRobot = Math.max(top, bottom);
+        min = Math.min(min, secondRobot);
+    }
+    return min;
+}
+
+// Main function to calculate the minimum result
+function gridGame(grid) {
```
Comment (aakhtar3) [inline javascript/2017-grid-game.js:31]: Move main function to the top of file.

File: javascript/2017-grid-game.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/**
+ * https://leetcode.com/problems/grid-game/
+ * Linear
+ * Time O(n) | Space O(n)
+ * @param {number[][]} grid
+ * @return {number}
+ */
+
+// Calculate the prefix sums for a given row in the grid
```
Comment (aakhtar3) [inline javascript/2017-grid-game.js:9]: Remove redundant comments

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/2017-grid-game.js (hunk 3 above)
Comment (aadil42) [reply javascript/2017-grid-game.js:9]: I see someone has added a better solution to this problem [here](https://github.com/neetcode-gh/leetcode/blob/main/javascript/2017-grid-game.js). I'm closing the PR.

**Comment by aadil42** [comment]
I see someone has added a better solution to this problem [here](https://github.com/neetcode-gh/leetcode/blob/main/javascript/2017-grid-game.js). I'm closing the PR.

**Pull request closed without merging**